### PR TITLE
Persons

### PR DESCRIPTION
**Pull request by stamp**

### CODE REVIEW

**Comment by stamp** [comment]
stamp_2020-08-10_124845
stamp_2020-08-10_124833

File: nodes/stampzilla-server/handlers/insecurewebsockethandler.go [hunk 1]
```diff
@@ -79,6 +80,8 @@ func (wsh *insecureWebsocketHandler) Connect(s interfaces.MelodySession, r *http
 		UUID:    wsh.Config.UUID,
 		TLSPort: wsh.Config.TLSPort,
 		Port:    wsh.Config.Port,
+		Init:    wsh.Store.CountAdmins() < 1,
+		Login:   helpers.IsPrivateIP(r.RemoteAddr),
```
Comment (jonaz) [inline nodes/stampzilla-server/handlers/insecurewebsockethandler.go:84]: Login == true means what?

File: nodes/stampzilla-server/models/connection.go [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 
+import "github.com/olahol/melody"
+
 type Connection struct {
 	Type       string                 `json:"type"`
 	RemoteAddr string                 `json:"remoteAddr"`
 	NodeUuid   string                 `json:"nodeUuid,omitEmpty"`
 	Attributes map[string]interface{} `json:"attributes"`
+
+	Session *melody.Session `json:"-"`
```
Comment (jonaz) [inline nodes/stampzilla-server/models/connection.go:11]: Is this used anywhere? searched for `.Session` in this PR and found nothing?

File: nodes/stampzilla-server/handlers/insecurewebsockethandler.go (hunk 1 above)
Comment (stamp) [reply nodes/stampzilla-server/handlers/insecurewebsockethandler.go:84]: Its probably a bad name but its used in the webui to know if it should show the login box or an error message saying "Only client certificates are accepted as login when connecting from the internet"

File: nodes/stampzilla-server/models/connection.go (hunk 2 above)
Comment (stamp) [reply nodes/stampzilla-server/models/connection.go:11]: I found a couple of places it was used in `webserver/webserver.go`
stamp_2020-09-04_100406
and
stamp_2020-09-04_100349

**Review by jonaz — APPROVED** [review]

**Comment by jonaz** [comment]
@stamp better late than never ;)  review done!

File: nodes/stampzilla-server/models/persons/person.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	State devices.State `json:"state"`
+}
+
+type PersonWithPassword struct {
+	Person
+	Password string `json:"password"`
+}
+
+type PersonWithPasswords struct {
+	PersonWithPassword
```
[the remark this answers is not in the document]
Comment (stamp) [reply nodes/stampzilla-server/models/persons/person.go:30]: Its to be able to json marshal/unmarshal a couple of different scenarios with the user and password.

Store users in users.json: username + password as an hash `PersonWithPassword`
Login request from client: username + password in clear text `PersonWithPassword`
Update password request from client: username + password + repeat password in clear text `PersonWithPasswords`
List users: only username `Person`

But I suppose the `PersonWithPasswords` one could have a better name... I didn't have a good plan when I wrote that part. Just solved one issue at the time.

**Comment by stamp** [comment]
> @stamp better late than never ;) review done!

yea, it was a big one so its not that easy to find the time needed to review it

File: nodes/stampzilla-server/models/persons/person.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonaz) [reply nodes/stampzilla-server/models/persons/person.go:30]: Ah, we dont want to leak the password for example :)  oherwise ,omitempty would have been enough.

**Pull request merged**